python/gstaichi/lang/_fast_caching/args_hasher.py

```diff
@@ -57,18 +60,21 @@ def stringify_obj_type(obj: Any) -> str | None:
     if isinstance(obj, MatrixField):
         return f"[fm-{obj.m}-{obj.n}-{obj.snode._id}-{obj.dtype}-{obj.shape}]"
     if dataclasses.is_dataclass(obj):
-        return dataclass_to_repr(obj)
+        return dataclass_to_repr(path, obj)
     if is_data_oriented(obj):
         child_repr_l = []
         for k, v in obj.__dict__.items():
-            _child_repr = stringify_obj_type(v)
+            _child_repr = stringify_obj_type(path + (k,), v)
             if _child_repr is None:
-                print("not representable child", k, type(v))
+                print("not representable child", k, type(v), "path", path)
                 return None
             child_repr_l.append(f"{k}: {_child_repr}")
         return ", ".join(child_repr_l)
-    if arg_type in [int, float, np.float32, np.float64, np.int32, np.int64, bool, np.bool]:
+    if arg_type in [int, float, np.float32, np.float64, np.int32, np.int64, bool]:
         return str(arg_type)
+    if arg_type is np.bool_:
+        # np is deprecating bool. Treat specially/carefully
+        return "np.bool_"
```

Comment:
`np.bool_` is not deprecated by Numpy and will never be, so why are you deprecating it? `np.bool` is deprecated though.

Reply:
It doesnt say we are deprecating bool_. It says that np is deprecating bool.

What makes you feel that we are deprecating `bool_`?